File: perl/functions/hash_operations.pl
#!/usr/bin/env perl
use strict;
use warnings;

my %person = (
    name => 'John',
    age => 30,
    city => 'New York'
);

print "Person:\n";
foreach my $key (keys %person) {
    print "  $key: $person{$key}\n";
}

print "\nName: $person{name}\n";
print "Age: $person{age}\n";

$person{email} = 'john@example.com';
print "\nAfter adding email:\n";
foreach my $key (keys %person) {
    print "  $key: $person{$key}\n";
}

if (exists $person{name}) {
    print "\nName exists: $person{name}\n";
}

delete $person{email};
print "\nAfter deleting email:\n";
foreach my $key (keys %person) {
    print "  $key: $person{$key}\n";
}

my @keys = keys %person;
print "\nKeys: @keys\n";

my @values = values %person;
print "Values: @values\n";

File: ruby/functions/hash_operations.rb
#!/usr/bin/env ruby

person = {
  name: 'John',
  age: 30,
  city: 'New York'
}

puts "Person: #{person}"
puts "Name: #{person[:name]}"
puts "Age: #{person[:age]}"
puts "Keys: #{person.keys}"
puts "Values: #{person.values}"

person[:email] = 'john@example.com'
puts "After adding email: #{person}"

if person.key?(:name)
  puts "Name exists: #{person[:name]}"
end

puts "\nIterating over hash:"
person.each do |key, value|
  puts "  #{key}: #{value}"
end

person.delete(:email)
puts "After deleting email: #{person}"

merged = person.merge(country: 'USA', state: 'NY')
puts "Merged: #{merged}"
